BackofficeBundle/Resources/public/coffee/page/AreaView.coffee

```diff
@@ -68,29 +68,63 @@ AreaView = OrchestraView.extend(
     ))
     @subBlocks.addClass (if @options.area.get("bo_direction") is "h" then "bo-row" else "bo-column")
 
-  sendBlockData: (event)->
-    event.stopImmediatePropagation() if event.stopImmediatePropagation
-    ul = $(event.target)
-    refreshUl ul
-    blocks = ul.children()
-    blockData = []
-    for block in blocks
-      info = $('div[data-block-type]', block)
-      if info.length > 0
-        if info.data('node-id') != '' && info.data('block-id') != '' 
-          blockData.push({'node_id' : info.data('node-id'), 'block_id' : info.data('block-id')})
-        else
-          blockData.push({'component' : info.data('block-type')})
-    mustRefresh = !! ul.find(".newly-inserted").length > 0
+  moveBlock: (event, ui) ->
+    event.stopPropagation()
+    areaFrom = $(event.currentTarget)
+    areaTo = $(ui.item.context).parent()
+    if (areaTo.find('.newly-inserted').length)
+      @addBlockToArea areaTo
+      return
+    if (ui.sender)
+      @moveBlockToNewArea ui.sender, areaTo
+    else if (areaFrom[0] == areaTo[0])
+      @moveBlockInSameArea areaFrom
```

Comment:
If ui.sender is undefined and areaFrom[0] != areaTo[0] in which case are we?
Could you do :
areaFrom = ui.sender || $(event.currentTarget)
if (areaFrom[0] == areaTo[0])
    @moveBlockInSameArea areaFrom
else
    @moveBlockToNewArea areaFrom, areaTo